source/mesh_deformation/interface.cc

```diff
@@ -0,0 +1,639 @@
+/*
+  Copyright (C) 2011 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/mesh_deformation/interface.h>
+#include <aspect/simulator.h>
+#include <aspect/global.h>
+
+#include <deal.II/dofs/dof_renumbering.h>
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/mapping_q1_eulerian.h>
+
+#include <deal.II/lac/sparsity_tools.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+
+
+namespace aspect
+{
+  namespace MeshDeformation
+  {
+    template <int dim>
+    Interface<dim>::~Interface ()
+    {}
+
+    template <int dim>
+    void
+    Interface<dim>::initialize ()
+    {}
+
+    template <int dim>
+    void
+    Interface<dim>::update ()
+    {}
+
+    template <int dim>
+    void
+    Interface<dim>::
+    declare_parameters (ParameterHandler &)
+    {}
+
+
+    template <int dim>
+    void
+    Interface<dim>::parse_parameters (ParameterHandler &)
+    {}
+
+
+    template <int dim>
+    MeshDeformationHandler<dim>::MeshDeformationHandler (Simulator<dim> &simulator)
+      : sim(simulator),  // reference to the simulator that owns the MeshDeformationHandler
+        free_surface_fe (FE_Q<dim>(1),dim), // Q1 elements which describe the mesh geometry
+        free_surface_dof_handler (sim.triangulation)
+    {}
+
+    template <int dim>
+    MeshDeformationHandler<dim>::~MeshDeformationHandler ()
+    {
+      // Free the Simulator's mapping object, otherwise
+      // when the MeshDeformationHandler gets destroyed,
+      // the mapping's reference to the mesh displacement
+      // vector will be invalid.
+      sim.mapping.reset();
```

Comment:
Who actually destroys the mesh deformation handler? Shouldn't that place be in charge of resetting the variable?